### PR TITLE
Standardize API error response.

### PR DESCRIPTION
**Pull request by fmassot**
### Description

Currently, the body of API error responses can be very different. 
It can be of the form 
```json
{
  "error": "error message that corresponds to error.to_string()",
}
```

or any errors that are JSON serializable. For example, with a `SearchError::InvalidQuery`, we are getting:

```json
{
  "InvalidQuery": "Invalid query: a {message}"
}
```

The status code can be retrieved from the HTTP response.

I would like to make the following changes: make all errors return a JSON following this format:

```json
{
  "message": "typically error.to_string()",
}
```

I have a preference for the field "message" rather than "error".

### CODE REVIEW

**Comment by fmassot** [comment]
@guilload I did not find an elegant way to use once and for all the `make_response` or the extraction of the format in the query params.

There are several reasons why I would not push for it now:

- we need to return the proper HTTP code 201 for "create" endpoints. This means we should return the result and a status code. `make_response` is not sufficient
- each handler currently returns `impl Reply` which is conv

File: quickwit/quickwit-serve/src/format.rs [hunk 1]
```diff
@@ -55,49 +58,69 @@ impl Serialize for Format {
...
-#[derive(Serialize)]
-pub(crate) struct FormatError {
-    #[serde(skip_serializing)]
-    pub code: ServiceErrorCode,
-    pub error: String,
+/// This struct represents a QueryString passed to
+/// the REST API.
+#[derive(Deserialize, Debug, Eq, PartialEq, utoipa::IntoParams)]
+#[into_params(parameter_in = Query)]
+pub(crate) struct FormatQueryString {
```
Comment (guilload) [inline quickwit/quickwit-serve/src/format.rs:65]: `Format` should be `BodyFormat`.

Comment (guilload) [inline quickwit/quickwit-serve/src/format.rs:65]:
```suggestion
struct FormatQueryString {
```

File: quickwit/quickwit-serve/src/format.rs [hunk 2]
```diff
@@ -55,49 +58,69 @@ impl Serialize for Format {
...
+    pub format: Format,
 }
 
-impl ToString for FormatError {
-    fn to_string(&self) -> String {
-        self.error.clone()
-    }
+pub(crate) fn extract_format_from_qs() -> impl Filter<Extract = (Format,), Error = Rejection> + Clone
+{
+    serde_qs::warp::query::<FormatQueryString>(serde_qs::Config::default()).and_then(
```
Comment (guilload) [inline quickwit/quickwit-serve/src/format.rs:73]:
```rust
serde_qs::warp::query::<FormatQueryString>(serde_qs::Config::default())
        .map(|format_qs: FormatQueryString| format_qs.format)
```

**Review by guilload — APPROVED** [review]

**Pull request merged**